cmd/kube-controller-manager/app/controllermanager.go

```diff
@@ -292,7 +292,8 @@ func (s *CMServer) Run(_ []string) error {
 	}
 
 	if s.AllocateNodeCIDRs {
-		if cloud == nil {
+		// TODO: Pipe this as a command line flag that corresponds to overlay==true
+		if cloud == nil || true {
```

Comment:
Also I uploaded this pr with flannel on by default because I wanted the e2e test run, I'm currently plumbing the options to configure --network-provider=X